Source/Core/DolphinQt/Achievements/AchievementSettingsWidget.cpp

```diff
@@ -39,8 +39,9 @@ AchievementSettingsWidget::AchievementSettingsWidget(QWidget* parent) : QWidget(
     ToggleHardcore();
 }
 
-void AchievementSettingsWidget::UpdateData()
+void AchievementSettingsWidget::UpdateData(bool login_failed)
 {
+  m_common_login_failed->setVisible(login_failed);
```

Comment:
It would be useful to actually pass up the correct error here instead of just a boolean, so the user knows if the login failed because their password was wrong or because their internet connection failed or whatever.

Reply:
Also the error message should probably stay visible until another login attempt is made instead of disappearing on the next possibly unrelated callback.